pyscf_ipu/nanoDFT/nanoDFT.py

```diff
@@ -110,55 +110,54 @@ def _nanoDFT(state, opts, mol):
     return log["matrices"], H_core, log["energy"]
 
 
-OrbitalVector = Float[Array, "num_orbitals"]
-OrbitalMatrix = Float[Array, "num_orbitals num_orbitals"]
-TwoOrbitalMatrix = Float[Array, "num_orbitals num_orbitals num_orbitals num_orbitals"]
-Grid = Float[Array, "4 grid_size num_orbitals"]
+FloatN = Float[Array, "N"]
+FloatNxN = Float[Array, "N N"]
+FloatNxNxNxN = Float[Array, "N N N N"]
+Grid = Float[Array, "4 grid_size N"]
 
 @chex.dataclass
 class IterationState:
-    """State tensors used during self-consistent field iterations
+    """State tensors used during self-consistent field (SCF) iterations
 
-    We use the following type annotations where the dimension N is the number
-    used in the linear combination of atomic orbitals (LCAO) basis set:
-
-        OrbitalVector [N] (float): Used to store the electron occupation mask
-        OrbitalMatrix [N, N] (float): Used for storing the one-electron integrals and density matrix.
-        TwoOrbitalMatrix [N, N, N, N] (float): 4-d matrix representing the two-electron
-            repulsion integrals.
-        Grid [4, grid_size, num_orbitals] (float): Numerical grid used to evaluate the
+        FloatN: Vector used to store the electron occupation mask.
+        FloatNxN: Square matrix used for storing the one-electron integrals and density matrix.
+        FloatNxNxNxN: 4-d matrix representing the two-electron repulsion integrals.
+        Grid [4, grid_size, N] (float): Numerical grid used to evaluate the
             exchange-correlation energy integral.
         
 
     Attributes:
         E_nuc (float): Energy of the nuclear-nuclear electrostatic interactions.
-        density_matrix (OrbitalMatrix): Electron density in the LCAO basis set.
-        kinetic (OrbitalMatrix): Kinetic energy integrals in the LCAO basis set.
-        nuclear (OrbitalMatrix): nuclear attraction integrals in the LCAO basis set.
-        O (OrbitalMatrix): Overlap integrals in the LCAO basis set.
+        density_matrix (FloatNxN): Electron density in the LCAO basis set.
+        kinetic (FloatNxN): Kinetic energy integrals in the LCAO basis set.
+        nuclear (FloatNxN): nuclear attraction integrals in the LCAO basis set.
+        O (FloatNxN): Overlap integrals in the LCAO basis set.
         grid_AO (Grid): Numerical grid used to evaluate the exchange-correlation energy integral.
-        ERI (TwoOrbitalMatrix): Two-electron repulsion integrals in the LCAO basis set.
+        ERI (FloatNxNxNxN): Two-electron repulsion integrals in the LCAO basis set.
         grid_weights (Array): Weights associated with the grid_AO
-        mask (OrbitalVector): Orbital occupation mask.
-        input_floats (Array):
-        input_ints (Array):
-        L_inv (OrbitalMatrix):
-        diis_history (Array): 
+        mask (FloatN): Orbital occupation mask.
+        input_floats (Array): Supplementary vector of floats for ERI evaluation with libcint
```

Comment:
Shape of this array?

Reply:
I'm not actually sure and I think the only way to know would be to run the code in a debugger.  Would be nice to document this in the type/shape annotation.